toplevel/genprintval.ml

```diff
@@ -568,10 +568,23 @@ module Make(O : OBJ)(EVP : EVALPATH with type valu = O.t) = struct
            identifier contained in the exception bucket *)
         if not (EVP.same_value slot (EVP.eval_address addr))
         then raise Not_found;
+        let type_params =
+          match (Ctype.repr cstr.cstr_res).desc with
+            Tconstr (_,params,_) ->
+             params
+          | _ -> assert false
+        in
+        let args=
+          List.map
+            (function ty ->
+                       try Ctype.apply env type_params ty ty_list with
+                         Ctype.Cannot_apply -> abstract_type)
```

Comment:
Could this logic, which is used in several different places in the module now, be moved to its own auxiliary function?